### PR TITLE
Initial code

### PR DESCRIPTION
**Pull request by ehoogerbeets**
- implement code checkers for python resource checkers
    - f-string replacement params
    - f-string numbered params
    - legacy-style replacement params

### CODE REVIEW

**Comment by ehoogerbeets** [comment]
This code is ready to review as a working first plugin example. The checks will fail until i18nlint-common 1.2.0 is published

File: src/FStringMatchRule.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
+/*
+ * FStringMatchRule.js - implement a rule to match f-string substitution parameters
+ *
+ * Copyright © 2022 JEDLSoft
```
Comment (gouniLee) [inline src/FStringMatchRule.js]: 2023?

File: src/FStringMatchRule.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+// from https://peps.python.org/pep-0498/
+const fstringRegExp = /\{(\}\}|[^}])*?\}/g;
+
+/**
+ * @class Represent an i18nlint rule.
+ */
+class FStringMatchRule extends Rule {
+    constructor(options) {
+        super(options);
+        this.name = "resource-python-fstrings-match";
```
Comment (gouniLee) [inline src/FStringMatchRule.js:32]: regarding the rule name, are there specific rules? or anything acceptable?
and How about type? Is not necessary this info? i.e )resource-matcher, resource-source a

File: src/FStringMatchRule.js (hunk 1 above)
Comment (ehoogerbeets) [reply src/FStringMatchRule.js]: It was written last year at Christmas time... but it hasn't been published yet, so I could put 2023 on everything.

File: src/FStringMatchRule.js (hunk 2 above)
Comment (ehoogerbeets) [reply src/FStringMatchRule.js:32]: I should put that in the README.md for ilib-lint

There are no rules, but I was following some conventions I made up. For the rules I write, resource checkers start with "resource-" and source file checkers start with "source-". For resource checkers, I used the word "match" in there when checking both the source and target, "source" when checking only the source string, and "target" when checking only the target string.

File: src/FStringMatchRule.js (hunk 1 above)
Comment (ehoogerbeets) [reply src/FStringMatchRule.js]: updated

**Review by gouniLee — APPROVED** [review]

**Pull request merged**